components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/service/WebSubEventSubscriberImpl.java

```diff
@@ -70,90 +68,92 @@ public String getName() {
-
-            List<BasicNameValuePair> params = new ArrayList<>();
-            params.add(new BasicNameValuePair(HUB_CALLBACK, callbackUrl));
-            params.add(new BasicNameValuePair(HUB_MODE, operation));
-            params.add(new BasicNameValuePair(HUB_TOPIC, topic));
-            if (secret != null) {
-                params.add(new BasicNameValuePair(HUB_SECRET, secret));
-            }
-            httpPost.setEntity(new UrlEncodedFormEntity(params, StandardCharsets.UTF_8));
-
-            WebSubHubCorrelationLogUtils.triggerCorrelationLogForRequest(httpPost);
-            final long requestStartTime = System.currentTimeMillis();
-
-            try (CloseableHttpResponse response = (CloseableHttpResponse) clientManager.executeSubscriberRequest(
-                    httpPost)) {
-                int responseCode = response.getStatusLine().getStatusCode();
-                if (responseCode >= 500 && attempt < MAX_RETRIES) {
-                    attempt++;
-                    log.debug("Retrying subscription API call, attempt " + attempt + " for topic: " + topic);
-                    try {
-                        Thread.sleep(RETRY_DELAY_MS);
-                    } catch (InterruptedException ie) {
-                        Thread.currentThread().interrupt();
-                        throw handleServerException(ERROR_SUBSCRIBING_TO_TOPIC, ie);
-                    }
-                    continue;
-                }
-                handleSubscriptionResponse(response, httpPost, topic, operation, requestStartTime);
-                break;
-            } catch (IOException | WebSubAdapterException e) {
-                log.debug("Error subscribing to topic: " + topic + ". Error: " + e.getMessage(), e);
-                if (attempt < MAX_RETRIES) {
-                    attempt++;
-                    log.debug("Retrying subscription API call, attempt " + attempt + " for topic: " + topic);
-                    try {
-                        Thread.sleep(RETRY_DELAY_MS);
-                    } catch (InterruptedException ie) {
-                        Thread.currentThread().interrupt();
-                        throw handleServerException(ERROR_SUBSCRIBING_TO_TOPIC, ie);
-                    }
-                    continue;
-                }
-                throw handleServerException(ERROR_SUBSCRIBING_TO_TOPIC, e);
-            }
+        HttpPost httpPost = clientManager.createHttpPost(webSubHubBaseUrl, null);
+        httpPost.setHeader("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8");
+
+        List<BasicNameValuePair> params = new ArrayList<>();
+        params.add(new BasicNameValuePair(HUB_CALLBACK, callbackUrl));
+        params.add(new BasicNameValuePair(HUB_MODE, operation));
+        params.add(new BasicNameValuePair(HUB_TOPIC, topic));
+        if (secret != null) {
+            params.add(new BasicNameValuePair(HUB_SECRET, secret));
```

Comment:
ideally this secret should be resolved here. If you have passed the webhook this default gets pushed to fix that

Reply:
Didn't get this